int32 keyboard_case
---
int32 return